File: rangeSum.java
import java.util.Stack;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

public class rangeSum {
    public int rangeSumBST(TreeNode root, int low, int high) {
        Stack<TreeNode> stack = new Stack();
        if (root == null) return 0;
        int ans = 0;
        stack.push(root);
        while(!stack.isEmpty()) {
            TreeNode curr = stack.pop();
            if (curr.val < low && curr.right != null) stack.push(curr.right);
            else if (curr.val > high && curr.left != null) stack.push(curr.left);
            else if (curr.val >= low && curr.val <= high) {
                ans += curr.val;
                if (curr.left != null) stack.push(curr.left);
                if (curr.right != null) stack.push(curr.right);
            }
        }
        return ans;
    }
}

File: rangeSum.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}

var rangeSumBST = function(root, low, high) {
    if (root == null) return 0;
    const stack = [root];
    let ans = 0;
    while (stack.length != 0) {
        let curr = stack.shift();
        if (curr.val > low && curr.left != null) stack.push(curr.left);
        if (curr.val < high && curr.right != null) stack.push(curr.right);
        if (curr.val <= high && curr.val >= low) ans += curr.val;
    }
    return ans;
}